GoRogue.UnitTests/MapGeneration/PolygonAreaTests.cs

```diff
@@ -96,5 +175,188 @@ public void PolygonAlgorithmsMustBeOrderedTest(Lines.Algorithm algorithm)
 
             Assert.Throws<ArgumentException>(() => new PolygonArea(algorithm, p1, p2, p3));
         }
+
+        [Fact]
+        public void PolygonFromRectangle()
+        {
+            var rect = new Rectangle(0, 0, 15, 10);
+            var polygon = PolygonArea.Rectangle(rect);
+
+            //Each Corner is included in two boundaries, so total number of points is offset by 4
+            Assert.Equal(rect.Area + 4, polygon.Count);
+            Assert.Equal(rect.PerimeterPositions().Count() + 4, polygon.OuterPoints.Count);
+            Assert.Equal(new HashSet<Point>(rect.Expand(-1, -1).Positions()), new HashSet<Point>(polygon.InnerPoints));
+            _output.WriteLine(GetPolygonString(polygon));
+        }
+
+        [Fact]
+        public void PolygonParallelogram()
+        {
+            var polygon = PolygonArea.Parallelogram((30, 30), 15, 15);
+            Assert.Equal(4, polygon.Corners.Count);
+            Assert.Equal(196, polygon.InnerPoints.Count);
+            Assert.Equal(64, polygon.OuterPoints.Count);
+            _output.WriteLine(GetPolygonString(polygon));
+        }
+
+        private void AssertPolygonSidesAreEqual(PolygonArea polygon, bool isStar)
+        {
+            var outer = (MultiArea)polygon.OuterPoints;//Satan forgive me
```

Comment:
This works fine functionally; but it does beg a question; is it worth exposing `OuterPoints` as some (currently non-existant) `IReadOnlyMultiArea` interface, in order to enable a user to do this without relying on this internal?

Reply:
We should just expose it as a `MultiArea` if anything other than `IReadOnlyArea`. I think there are use cases for distinguishing the boundaries from one another in user code.

Reply:
I don't think exposing it as a `MultiArea` specifically is ideal, because `MultiArea` has a function that allows adding sub-areas; and we wouldn't want a user adding to the `SubAreas` of outer points themselves as it would pretty drastically change invariants the class can assume and/or cause lots of bugs.

If we're going to expose it, I think it would still need to be in a read-only form; but right now we don't have an appropriate read-only interface.  Adding one to GoRogue and implementing it seems like a task best accomplished in a separate PR, because it could have implications outside of just its usage in `PolygonArea`.

Created #239 to track, and specifically mentioned `OuterPoints` in `PolygonArea`.  Feel free to consider this resolved for the purposes of this PR.